### PR TITLE
Revert "Revert "wayland: mark as broken on darwin""

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

tl;dr: It is a mistake to port one package to macOS while breaking tons of others.

This PR reverts #214828, which misleadingly builds a stub Wayland package on macOS (see https://github.com/NixOS/nixpkgs/pull/214828#issuecomment-1418660135) and causes build failures on almost all packages marked as broken transitively by Wayland before.

The issue was observed at least two weeks ago, in #212123, when the maintainer discovered that `inherit (wayland.meta) platforms` didn't work as intended on macOS.

cc @alyssais @figsoda

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alyssais** [comment]
As I explained, a Wayland package on macOS is useful because of wayland-scanner (which is in the `bin` output, hence you not finding it in https://github.com/NixOS/nixpkgs/pull/214828#issuecomment-1418660135).  Wayland can be built for all Unixes to provide wayland-scanner, and it is necessary to have when cross-compiling to Linux.

Indeed `wayland.meta.platforms` does include Darwin, but this is correct, as the package does build for Darwin and contain useful functionality.  It does not include _all_ functionality, but that's not something that `meta.platforms` can encode.

In spite of this, if you still think this change is the right thing to do, then I would like to suggest we ask for third opinions from my co-maintainers of the Wayland package: @primeos @codeopel.

**Comment by wegank** [comment]
> (I'm also happy to go through and fix any incorrect occurrences on `wayland.meta.platforms` or similar — there seem to be a few but not that many. But I'd leave it for a few days, because I'm about to PR full FreeBSD support for our Wayland package, including the libraries, and that'll make it easier to test what the actually supported platforms for each compositor are: just Linux, or Linux+FreeBSD.)

I'm having an unofficial full-fledged Wayland package on macOS [in my repo](https://github.com/wegank/nur-packages/blob/main/pkgs/development/libraries/wayland/default.nix), so doing a simple comparison, I don't agree if what betweens _useful_ and _all_ functionality is the existence of libraries and headers in `wayland.dev`.

I'll be happy to test `epoll-shim` on macOS, if you decided to use the latest git version.

**Comment by alyssais** [comment]
> I'm having an unofficial full-fledged Wayland package on macOS [in my repo](https://github.com/wegank/nur-packages/blob/main/pkgs/development/libraries/wayland/default.nix)

That's cool! I hope you can upstream it, as that would resolve all our problems here. :)

> I'll be happy to test `epoll-shim` on macOS, if you decide to use the latest git version.

That's exactly what I'll be doing, because the latest git commit is a build fix from me. :P

**Comment by alyssais** [comment]
> I don't agree if what betweens _useful_ and _all_ functionality includes the non-existence of headers in `wayland.dev` and libraries in `wayland`:

I'm not entirely sure I understand what you mean here, but the headers should be built if and only if the libraries are, since otherwise the headers would be useless. The reason to have current Wayland be buildable for macOS and other Unixes is to have the wayland-scanner binary and associated .pc file available.

**Comment by alyssais** [comment]
@wegank would https://github.com/NixOS/nixpkgs/pull/214906 be an acceptable solution?

**Comment by wegank** [comment]
> @wegank would #214906 be an acceptable solution?

Thanks! I've also been trying to do something similar for the last hour, realizing finally that anything I can do should go to staging. So I think I'll close this PR now in favor of yours.

**Comment by alyssais** [comment]
Cool. Glad we've got a path forward. :)
